src/renderer/components/+preferences/preferences-navigation/extensions-preferences-navigation-item.injectable.ts

```diff
@@ -3,49 +3,54 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 import { getInjectable } from "@ogre-tools/injectable";
-import { preferenceNavigationItemInjectionToken } from "./preference-navigation-items.injectable";
-import routeIsActiveInjectable from "../../../routes/route-is-active.injectable";
 import { computed } from "mobx";
-import extensionsPreferenceItemsInjectable from "../extension-preference-items.injectable";
 import extensionPreferencesRouteInjectable from "../../../../common/front-end-routing/routes/preferences/extension/extension-preferences-route.injectable";
-import navigateToPreferenceTabInjectable from "./navigate-to-preference-tab.injectable";
+import navigateToExtensionPreferencesInjectable from "../../../../common/front-end-routing/routes/preferences/extension/navigate-to-extension-preferences.injectable";
+import { extensionRegistratorInjectionToken } from "../../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensRendererExtension } from "../../../../extensions/lens-renderer-extension";
+import currentPathParametersInjectable from "../../../routes/current-path-parameters.injectable";
+import routeIsActiveInjectable from "../../../routes/route-is-active.injectable";
+import { preferenceNavigationItemInjectionToken } from "./preference-navigation-items.injectable";
 
-const extensionsPreferencesNavigationItemInjectable = getInjectable({
+const extensionPreferencesNavigationItemRegistratorInjectable = getInjectable({
   id: "extension-preferences-navigation-item",
 
   instantiate: (di) => {
-    const preferenceItems = di.inject(
-      extensionsPreferenceItemsInjectable,
-    );
-
-    const navigateToPreferenceTab = di.inject(
-      navigateToPreferenceTabInjectable,
-    );
-
-    const route = di.inject(
-      extensionPreferencesRouteInjectable,
-    );
-
-    const routeIsActive = di.inject(
-      routeIsActiveInjectable,
-      route,
-    );
-
-    return {
-      id: "extensions",
-      label: "Extensions",
-      navigate: navigateToPreferenceTab(route),
-      isActive: routeIsActive,
-
-      isVisible: computed(
-        () => preferenceItems.get().length > 0,
-      ),
-
-      orderNumber: 70,
+    return (ext) => {
+      const extension = ext as LensRendererExtension;
```

Comment:
Why not like this? Is there something wrong with types in args?

```suggestion
    return (extension: LensRendererExtension) => {
```

Reply:
Unfortunately that doesn't work because the typing required are `(extension: LensExtension, extensionInstallationCount: number) => void`. This cast was previously done in `ExtensionLoader` but for now needs to be done in these functions.